### PR TITLE
Migrate to Compose

### PR DESCRIPTION
**Pull request by chris-horner**

### CODE REVIEW

File: app/src/debug/kotlin/codes/chrishorner/socketweather/data/DebugNetworkComponents.kt [hunk 1]
```diff
@@ -50,42 +46,30 @@ class DebugNetworkComponents(
...
-          message.startsWith('[') -> JSONArray(message).toString(2)
-          else -> message
-        }
-      } catch (e: JSONException) {
-        message
+  private val logger = HttpLoggingInterceptor { message ->
+    val formattedMessage: String = try {
+      when {
+        message.startsWith('{') -> JSONObject(message).toString(2)
+        message.startsWith('[') -> JSONArray(message).toString(2)
```
Comment (adamjcampbell) [inline app/src/debug/kotlin/codes/chrishorner/socketweather/data/DebugNetworkComponents.kt:53]: 😂 is whitespace technically a valid start character? *looks up JSON spec*

Comment (adamjcampbell) [reply app/src/debug/kotlin/codes/chrishorner/socketweather/data/DebugNetworkComponents.kt:53]: Seems to be the case 😭 may want to trim leading and trailing whitespace

**Review by adamdickinson — APPROVED** [review]
Android/Kotlin isn't my forte, but I can't find anything I can justifiably question.

Nice work! :100:

File: app/src/debug/kotlin/codes/chrishorner/socketweather/data/DebugNetworkComponents.kt (hunk 1 above)
Comment (chris-horner) [reply app/src/debug/kotlin/codes/chrishorner/socketweather/data/DebugNetworkComponents.kt:53]: Not worth the effort for debug logs on an API that should never return a result beginning with whitespace.

**Pull request merged**